src/runtime/pkg/katautils/utils_unix.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2022 NetEase Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package katautils
+
+import (
+	"fmt"
+	"github.com/kata-containers/kata-containers/src/runtime/pkg/device/config"
+	"golang.org/x/sys/unix"
+	"os"
+)
+
+func GetDeviceInfoByPath(devPath string) (config.DeviceInfo, error) {
+	var devInfo config.DeviceInfo
+	stat, err := os.Stat(devPath)
+	if err != nil {
+		return devInfo, fmt.Errorf("error stating device path: %w", err)
+	}
+
+	if !stat.IsDir() {
+		dev, err := DeviceFromPath(devPath)
+		if err != nil {
+			return devInfo, err
+		}
+		return *dev, nil
+	}
+
+	return devInfo, nil
```

Comment:
This returns a default value. And further be attached by sandbox, it this an expected result, just being curious.

Reply:
updated the case when the device path is a directory, please review again.